ably/types/options.py

```diff
@@ -42,12 +43,18 @@ def __init__(self, client_id=None, log_level=0, tls=True, rest_host=None, realti
         if environment is not None and realtime_host is not None:
             raise ValueError('specify realtime_host or environment, not both')
 
+        if endpoint is not None and (environment is not None or rest_host is not None or realtime_host is not None):
```

Comment:
This check should be before Line 40, since we plan to implement early returns/early checks if `endpoint` not null/None.